### ISSUE TITLE
[EMAGE]: Convert my ARKit blendshape coefficient to FLAME coefficient

### ISSUE THREAD

**Issue by fdyuandong** [PantoMatrix/PantoMatrix#130]
Hi Liu,
Excellent work! I want to convert my ARKit blendshape coefficient into FLAME coefficient. Could you provide the transformation matrix W? 
If not, how should I go about obtaining this W matrix?

**Comment by H-Liu1997** [comment]
thank you so much for the postivate feedback of EMAGE, you may consider to use this [transformation matrix](https://drive.google.com/file/d/1OV6OFYgj0vNuvgqBPLG0CfWQ5TgGWfID/view?usp=drive_link)

**Comment by fdyuandong** [comment]
Thanks for your quick reply!

**Comment by fdyuandong** [comment]
> thank you so much for the postivate feedback of EMAGE, you may consider to use this.

The dimensions of the transformation matrix is 103x51. If I want to generate a 100-dimensional vector similar to BAET2, should I just use the first 100 dimensions of the vector?

**Comment by H-Liu1997** [comment]
the additional 3 dims are axis angles of jaw open in FLAME

Best
Haiyang

**Comment by adonispeace** [comment]
Thank you for your generous sharing. I used it this way and found that it was not converted into the bs corresponding to arkit. Do you have any examples of use? thx！

``` python     
    n_shape = 300
    n_expr = 100
    model = FlameHead(n_shape, n_expr, add_teeth=False)

    shape = torch.zeros((51, n_shape), dtype=torch.float32)
    rotation = torch.zeros((51, 3), dtype=torch.float32)
    neck = torch.zeros((51, 3), dtype=torch.float32)
    eyes = torch.zeros((51, 6), dtype=torch.float32)
    translation = torch.zeros((51, 3), dtype=torch.float32)

    # update exp
    exp = np.eye(51)
    arkit_to_flame = exp @ M
    exp_flame = torch.from_numpy(arkit_to_flame[:, :100]).float()
    jaw = torch.from_numpy(arkit_to_flame[:, 100:]).float()

    ret_vals, _ = model(shape, exp_flame, rotation, neck, jaw, eyes, translation)
```

**Comment by MOr1s2** [comment]
@fdyuandong Sorry to bother you, how did you get the data in ARKit format?

**Comment by edhyah** [comment]
@fdyuandong How many images did you use to regress this matrix?

**Comment by JosephKKim** [comment]
@H-Liu1997 Hello, I know it's bit off from the issue subject, but is there any way to get FLAME parameters -> ARKit blendshapes?

**Comment by emlcpfx** [comment]
> [@H-Liu1997](https://github.com/H-Liu1997) Hello, I know it's bit off from the issue subject, but is there any way to get FLAME parameters -> ARKit blendshapes?

Did you learn a way to get ARKit blendshapes from FLAME params?

**Comment by henning101** [comment]
> thank you so much for the postivate feedback of EMAGE, you may consider to use this [transformation matrix](https://drive.google.com/file/d/1OV6OFYgj0vNuvgqBPLG0CfWQ5TgGWfID/view?usp=drive_link)

@H-Liu1997 Hi I would also be interested in the ARKit-FLAME conversion matrix but the link is not available anymore. Could you maybe provide a new link? Thank you so much in advance!

**Comment by sora158** [comment]
> thank you so much for the postivate feedback of EMAGE, you may consider to use this [transformation matrix](https://drive.google.com/file/d/1OV6OFYgj0vNuvgqBPLG0CfWQ5TgGWfID/view?usp=drive_link)

Hello, I found this link has expired, could you provide a new one please?

**Comment by j71200** [comment]
The file [ARkit2FLAME Weights](https://huggingface.co/datasets/H-Liu1997/BEAT2_Tools/blob/main/mat_final.npy) is a matrix M with shape (51, 103).

I want to convert ARkit blendshapes to FLAME.
For example, if my ARkit blendshapes are `x` with length 51 in this order #199 
I compute

`y = x @ M`

which gives `y` with length 103.

`y[100:103] = y[100:103] * 100  # Scale the jaw angles by 100 (you could use other scaling value)`

This y is then used as the FLAME blendshape.
So far, the facial expression looks not too weird.

**Comment by H-Liu1997** [comment]
> > thank you so much for the postivate feedback of EMAGE, you may consider to use this [transformation matrix](https://drive.google.com/file/d/1OV6OFYgj0vNuvgqBPLG0CfWQ5TgGWfID/view?usp=drive_link)
> 
> Hello, I found this link has expired, could you provide a new one please?

hello, I moved it to hugging face:
https://huggingface.co/datasets/H-Liu1997/BEAT2_Tools/blob/main/mat_final.npy
thank you!
